### DISCUSSION TITLE
Healthchecks.io cloud export/migrate to self-hosted

### DISCUSSION THREAD

**Discussion by sebestyeng** [healthchecks/healthchecks#631] [Q&A] [answered]
Hi,
I would like to know is there a way to export or migrate my projects and checks from a healthchecks.io to a self-hosted one?

**Comment by cuu508** [comment, marked as answer]
In terms of what's available right now, – you can clone checks from one project into another using Management API. [This page](https://healthchecks.io/docs/cloning_checks/) has a code snippet that clones checks from one project into another, you could adjust it to use a different `API_URL` for the target project.

It would be great to have a feature where users can export a snapshot of their accounts and import the snapshot elsewhere. This doesn't exist at the moment. I have a couple internal scripts that I once used to selectively restore data from backup (to restore an account that was deleted by accident), I will investigate if it is feasible to build something off them.

**Comment by sebestyeng** [comment]
Thank you for informative answer!

**Comment by sebestyeng** [comment]
If someone wants to use this option i've created a short script which copies checks from a source to a target healthchecks cloud or self-hosted server.
Please change the source and target URL variables to your needs!

```
import requests

SOURCE_API_URL = "https://healthchecks.io/api/v1/checks/"
TARGET_API_URL = "http://0.0.0.0:8000/api/v1/checks/"
SOURCE_PROJECT_READONLY_KEY = "SOURCE_API_KEY_READONLY"
TARGET_PROJECT_KEY = "TARGET_API_KEY"

r = requests.get(SOURCE_API_URL, headers={"X-Api-Key": SOURCE_PROJECT_READONLY_KEY})
for check in r.json()["checks"]:
    print("Cloning %s" % check["name"])
    requests.post(TARGET_API_URL, json=check, headers={"X-Api-Key": TARGET_PROJECT_KEY})
```
